lib/engines/hcktest/playlist.rb

```diff
@@ -82,6 +82,17 @@ def sort_by_duration
       @tests.sort_by! { |test| test['duration'] }
     end
 
+    def intersect_select_tests(select_test_names)
+      select_test_names_counted = select_test_names.tally
+
+      @tests = @tests.filter_map do |test|
+        if select_test_names_counted.key?(test['name'])
+          test['run_count'] = select_test_names_counted[test['name']] if @project.options.test.allow_test_duplication
```

Comment:
Assign `1` to `run_count` when not allowing test duplication. It's better to have a consistent data layout whether allowing test duplication or not.